ui/core-react/src/core-react/dialog/Dialog.tsx

```diff
@@ -386,3 +334,77 @@ export class Dialog extends React.Component<DialogProps> {
     }
   };
 }
+
+function DialogButton({
+  button,
+  primaryStyleType,
+  noCoreButtonClasses,
+}: {
+  button: DialogButtonDef;
+  primaryStyleType: ButtonProps["styleType"];
+  noCoreButtonClasses: boolean;
+}) {
+  const { translate } = useTranslation();
+
+  let buttonText = "";
+
+  let buttonClass = classnames(
+    !noCoreButtonClasses && "core-dialog-button",
+    !noCoreButtonClasses && `dialog-button-${button.type}`,
+    button.className
+  );
+  let usePrimaryStyleType = false;
+
+  switch (button.type) {
+    case DialogButtonType.OK:
+      buttonText = translate("dialog.ok");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      usePrimaryStyleType = true;
+      break;
+    case DialogButtonType.Retry:
+      buttonText = translate("dialog.retry");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      usePrimaryStyleType = true;
+      break;
+    case DialogButtonType.Yes:
+      buttonText = translate("dialog.yes");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      usePrimaryStyleType = true;
+      break;
+    case DialogButtonType.No:
+      buttonText = translate("dialog.no");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      break;
+    case DialogButtonType.Cancel:
+      buttonText = translate("dialog.cancel");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      break;
+    case DialogButtonType.Close:
+      buttonText = translate("dialog.close");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      break;
+    case DialogButtonType.Next:
+      buttonText = translate("dialog.next");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      usePrimaryStyleType = true;
+      break;
+    case DialogButtonType.Previous:
+      buttonText = translate("dialog.previous");
+      buttonClass = classnames(buttonClass, button.buttonStyle);
+      usePrimaryStyleType = true;
+      break;
+  }
+
+  buttonText = button.label ? button.label : buttonText;
```

Comment:
```suggestion
  buttonText = button.label ?? buttonText;
```

Reply:
I had this change initially, but wasn't sure about empty string, since operators will returns different results.

Reply:
Replaced with logical OR.